app/code/Magento/Sales/Model/AdminOrder/Create.php

```diff
@@ -2037,7 +2038,30 @@ protected function _validate()
      */
     protected function _getNewCustomerEmail()
     {
-        return $this->getData('account/email');
+        $emailrequired = $this->_scopeConfig->getValue(
```

Comment:
Method becomes difficult to read.
Could move parts of logic to private methods?

Reply:
@swnsma 

Please explain in more details.

Reply:
@solwininfotech,
You can separate the method into smaller methods:
- getIsEmailRequired - which extract config value from `XML_PATH_EMAIL_REQUIRED_CREATE_ORDER`
- generateEmail - which return generated email (the code inside `if (empty($email))` statement.

It should make the code more readable.

Reply:
@swnsma 

Done.